### PR TITLE
Webhook: Git provider

### PR DESCRIPTION
**Pull request by shivani170**
# Description

Screenshot 2021-07-21 at 2 39 44 PM

### CODE REVIEW

File: src/components/gitProvider/GitProvider.tsx [hunk 1]
```diff
@@ -1,41 +1,135 @@
...
+    const [providerList, setProviderList] = useState([])
+    const [hostList, setHostList] = useState([])
+    const [hostListOption, setHostListOption] = useState([])
+    const [isPageLoading, setIsPageLoading] = useState(true)
+    const [isErrorLoading, setIsErrorLoading] = useState(false)
+    const [errors, setErrors] = useState([])
+    async function getInitData() {
+        try {
+            const { result: providers = [] } = await getGitProviderList()
+            const { result: hosts = [] } = await getGitHostList()
```
Comment (pghildiyal) [inline src/components/gitProvider/GitProvider.tsx:36]: use [result1, result2] = Promise.all([async1(), async2()]); for line no 35 and 36

File: src/components/gitProvider/GitProvider.tsx [hunk 2]
```diff
@@ -1,41 +1,135 @@
...
+            providers.sort((a, b) => a.name.localeCompare(b.name))
+            hosts.sort((a, b) => a.name.localeCompare(b.name))
+            let hostOptions = hosts.map(host => {
```
Comment (pghildiyal) [inline src/components/gitProvider/GitProvider.tsx:39]: why do we need both hosts and hostOptions? one can be created from other

File: src/components/gitProvider/GitProvider.tsx [hunk 3]
```diff
@@ -1,41 +1,135 @@
...
+        } catch (error) {
+            showError(error)
+            setErrors(error)
+            setIsErrorLoading(true)
+        } finally {
+            setIsPageLoading(false)
+        }
+    }
+
+    async function getHostList() {
```
Comment (pghildiyal) [inline src/components/gitProvider/GitProvider.tsx:57]: seems to have duplication with initdata

File: src/components/gitProvider/GitProvider.tsx [hunk 4]
```diff
@@ -1,41 +1,135 @@
...
+            const { result: hosts = [] } = await getGitHostList();
+            hosts.sort((a, b) => a.name.localeCompare(b.name))
+            setHostList(hosts)
+        } catch (error) {
+            showError(error)
+            setIsErrorLoading(true)
+        }
+    }
+
+    async function getProviderList() {
```
Comment (pghildiyal) [inline src/components/gitProvider/GitProvider.tsx:68]: seems to have duplication with init data

File: src/components/gitProvider/GitProvider.tsx [hunk 5]
```diff
@@ -1,41 +1,135 @@
...
+                    reload={getInitData}
+                />
+            })}
+            {/* {[{ id: null, name: "", active: true, url: "", authMode: "ANONYMOUS" }].concat(result && Array.isArray(result.result) ? result.result : []).sort((a, b) => a.name.localeCompare(b.name)).map(git => <CollapsedList {...git} key={git.id || Math.random().toString(36).substr(2, 5)} reload={reload} />)} */}
+        </section>
     )
 }
 
-function CollapsedList({ id, name, active, url, authMode, accessToken = "", userName = "", password = "", reload, ...props }) {
+function CollapsedList({ id, name, active, url, authMode, gitHostId, accessToken = "", userName = "", password = "", reload, hostListOption, hostList, getHostList, getProviderList, providerList, ...props }) {
```
Comment (pghildiyal) [inline src/components/gitProvider/GitProvider.tsx:132]: too many input arguments should be avoided.. they should be wrapped up in an object in such cases

File: src/components/gitProvider/GitProvider.tsx [hunk 6]
```diff
@@ -133,19 +234,37 @@ function GitForm({ id = null, name = "", active = false, url = "", authMode = nu
...
             ...(state.auth.value === 'USERNAME_PASSWORD' ? { username: customState.username.value, password: customState.password.value } : {}),
             ...(state.auth.value === 'ACCESS_TOKEN' ? { accessToken: customState.accessToken.value } : {})
         }
+
         const api = id ? updateGitProviderConfig : saveGitProviderConfig
         try {
             setLoading(true)
-            const { result } = await api(payload, id);
+            await api(payload, id);
             await reload();
```
Comment (pghildiyal) [inline src/components/gitProvider/GitProvider.tsx:268]: [result1, result2] = Promise.all([async1(), async2()]); can be used to combine line no 267 and 268

**Comment by shivani170** [comment]
Image building on every save

**Pull request closed without merging**